data/transactions/logic/README_in.md

```diff
@@ -200,32 +217,47 @@ _available_.
  * The `txn.Sender`, `global CurrentApplicationID`, and `global
    CurrentApplicationAddress` are _available_.
 
- * Prior to v4, all assets were considered _available_ to the
-   `asset_holding_get` opcode, and all applications were _available_
-   to the `app_local_get_ex` opcode.
-
- * Since v6, any asset or contract that was created earlier in the
-   same transaction group (whether by a top-level or inner
-   transaction) is _available_. In addition, any account that is the
-   associated account of a contract that was created earlier in the
-   group is _available_.
-
- * Since v7, the account associated with any contract present in the
-   `txn.ForeignApplications` field is _available_.
-   
- * Since v9, there is group-level resource sharing. Any resource that
-   is available in _some_ top-level transaction in a transaction group
-   is available in _all_ v9 or later application calls in the group,
-   whether those application calls are top-level or inner.
-   
+ * In pre-v4 applications, all holdings are _available_ to the
+   `asset_holding_get` opcode, and all locals are _available_ to the
+   `app_local_get_ex` opcode if the *account* of the resource is
+   _available_.
+
+ * In v6 and later applications, any asset or application that was
+   created earlier in the same transaction group (whether by a
+   top-level or inner transaction) is _available_. In addition, any
+   account that is the associated account of a contract that was
+   created earlier in the group is _available_.
+
+ * In v7 and later applications, the account associated with any
+   contract present in the `txn.ForeignApplications` field is
+   _available_.
+
+ * in v4 and above applications, Holdings and Locals are _available_
+   if, both components of the resource are available according to the
+   above rules.
+
+ * In v9 and later applications, there is group-level resource
+   sharing. Any resource that is available in _some_ top-level
+   transaction in a transaction group is available in _all_ v9 or
+   later application calls in the group, whether those application
+   calls are top-level or inner.
+
+ * v9 and later applications may use the `txn.Access` list instead of
```

Comment:
Does the top level transaction alone determine the availability in all inners? i.e. what if an inner involves a pre-v9 application, etc...

Reply:
It's the version that's running that matters. So if a v5 calls a v9, the v9 will have access to group shared resources.

If v9 calls v7, it must pass the resources down in foreign-array fields when it creates the inner transaction, because it will only have access to locally available stuff.